Replicator/Worker.hh

```diff
@@ -247,13 +247,7 @@ namespace litecore { namespace repl {
         // then, it returns a pair of {collectionIndex, errorSlice} with the post-conditions:
         //   errorSlice != nullslice || (0 <= collectionIndex < _options->workingCollectionCount()
         // 'errorSlice' describes the nature of the violation.
-        // It may be used in two contexts. When handling a reqeust, the collection index must
-        // be in the range of 0, inclusive, and workingCollectionCount(), exclusive.
-        // When handling a response, we further check it against the collection index of
-        // the original request. This context is indicated by the second argument of
-        // CollectionIndex being not kNotCollectionIndex.
-        std::pair<CollectionIndex, slice> checkCollectionOfMsg(const blip::MessageIn& msg,
-                                                               CollectionIndex) const;
+        std::pair<CollectionIndex, slice> checkCollectionOfMsg(const blip::MessageIn& msg) const;
```

Comment:
There is nothing wrong with the functionality, but again, I want to point out that I think we should avoid a function that returns two thing like this. There are standard ways of returning an error such as using exception or using an out error parameter. Also you could break the function into two as well if sometime the CollectionIndex is not used such as :

`bool validateCollectionIndex(CollectionIndex index, C4Error* error)`
`int getCollectionIndex(const blip::MessageIn& msg,  C4Error* error)` // Will call validateCollectionIndex() 

For error, you can use anything that we are using in the other places in LiteCore for consistency. I don't know if there is a case before that we return only error message. If so, you can follow that pattern. With C4Error, you can set the domain and error message.

Reply:
Got you. I chose this way for the reason of expediency because it's a private utility method. In fact, we have many methods returning pairs in Core, if you search "auto ["
The reason I return an error msg is that there are two difference uses, one will use the message in Error, and the other will use it as part of response back to remote.

Reply:
> Got you. I chose this way for the reason of expediency because it's a private utility method. In fact, we have many methods returning pairs in Core, if you search "auto ["

I checked that but those are differences as they don't returns both result and error in pair.